hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/EntityStakeRepository.java

```diff
@@ -69,25 +68,31 @@ select lower(timestamp_range) as timestamp from entity_history where id = 800
           limit 1
         ), entity_state as (
           select
-            decline_reward,
             id,
             staked_account_id,
             staked_node_id,
             stake_period_start
           from entity, end_period
-          where deleted is not true and type in ('ACCOUNT', 'CONTRACT') and timestamp_range @> end_period.consensus_timestamp
+          where id = 800 or
+            (deleted is not true and type in ('ACCOUNT', 'CONTRACT') and timestamp_range @> end_period.consensus_timestamp
+                and
+              ((decline_reward is false and (staked_node_id <> -1 or staked_account_id <> 0)) or (decline_reward is true and staked_account_id <> 0))
+            )
           union all
           select *
           from (
             select
               distinct on (id)
-              decline_reward,
               id,
               staked_account_id,
               staked_node_id,
               stake_period_start
             from entity_history, end_period
-            where deleted is not true and type in ('ACCOUNT', 'CONTRACT') and timestamp_range @> end_period.consensus_timestamp
+            where id = 800 or
```

Comment:
can't include 800 in both subqueries (`entity` and `entity_history`), we may end up having two rows for 800 since we do a `union all`

change it to
```suggestion
            where id <> 800 and
```

Reply:
Got it thanks!